packages/mobile/src/in-house-liquidity/index.ts

```diff
@@ -5,51 +5,69 @@ import { generateKeys } from '@celo/utils/lib/account'
 
 export const createPersonaAccount = async (accountMTWAddress: string): Promise<Response> => {
   const body = { accountMTWAddress }
-  const headers = await getAuthAndDateHeaders(
-    'post persona/account/create',
-    accountMTWAddress,
-    JSON.stringify(body)
-  )
-  return fetch(`${networkConfig.inhouseLiquditiyUrl}/persona/account/create`, {
+  return signAndFetch('/persona/account/create', accountMTWAddress, {
     method: 'POST',
     headers: {
       'Content-Type': 'application/json',
-      ...headers,
     },
     body: JSON.stringify(body),
   })
 }
 
 // Helper Functions
 
-const getPrivateKey = async (accountMTWAddress: string): Promise<string> => {
+const getKeys = async (accountMTWAddress: string) => {
   const mnemonic = await getStoredMnemonic(accountMTWAddress)
```

Comment:
I think the input for getStoredMnemonic is supposed to be an EOA, not an MTW address

Reply:
in `BackupPhrase.tsx` the same function is used, but an EOA address is supplied (rather than an MTW)

Reply:
I tried it just now with both. Using MTW address works

Using EOA address returns a different privateKey 😮 which IHL then says is not a signature that matches the call.

Reply:
We shouldn't use the mnemonic this way.

The EOA or DEK private keys are already stored securely and should be accessed via the `wallet` interface (see [`getWallet`](https://github.com/valora-inc/wallet/blob/8a2718ba8b7b74540d93c98cf8929f83364e144c/packages/mobile/src/web3/contracts.ts#L170)
Its `signTypedData` function should probably be preferred.

Not entirely familiar with IHL, but the DEK may be more appropriate to use.

Anyway looks like we need to discuss to share more context.

Reply:
Following, would like to learn more about this when you guys discuss